R/summarize_weighted_production.R

```diff
@@ -269,7 +268,9 @@ add_technology_share <- function(data) {
       .data$name_ald,
       .data$region
     ) %>%
-    mutate(technology_share = .data$production / sum(.data$production)) %>%
+    mutate(prod_total_company=sum(.data$production),
+           tech_share=.data$production / sum(.data$production),
+           technology_share = ifelse(prod_total_company!=0,tech_share,0)) %>%
```

Comment:
NIT: Since we don't use `prod_total_company` again in the pipeline, it's better to set it to a temporary variable (prefixed by a `.`), and then set that variable back to `NULL` at the end of `mutate()`, so we don't pollute the dataset with unnecessary columns: 
``` r 
mutate(
   .x = sum(.data$production),
   technology_share = ifelse(
      .x == 0,
      0,
      .data$production / .x
   ),
   .x = NULL
)
```